stats/exporter/stackdriver/stackdriver.go

```diff
@@ -0,0 +1,242 @@
+	timestamp "github.com/golang/protobuf/ptypes/timestamp"
+	"google.golang.org/api/option"
+	"google.golang.org/api/support/bundler"
+	distributionpb "google.golang.org/genproto/googleapis/api/distribution"
+	metricpb "google.golang.org/genproto/googleapis/api/metric"
+	monitoredrespb "google.golang.org/genproto/googleapis/api/monitoredres"
+	monitoringpb "google.golang.org/genproto/googleapis/monitoring/v3"
+)
+
+// Exporter exports stats to the Stackdriver Monitoring.
+type Exporter struct {
+	// ProjectID is the identifier of the Stackdriver
+	// project the user is uploading the stats data to.
+	ProjectID string
+
+	// OnError is the hooked to be called when there is
+	// an error occured when uploading the stats data.
+	// If no custom hook is set, errors are logged.
+	// Optional.
+	OnError func(err error)
+
+	// ClientOptions are additional options to be passed
+	// to the underlying Stackdriver Monitoring API client.
+	// Optional.
+	ClientOptions []option.ClientOption
+
+	// ExportDelayThreshold determines the max amount of time
+	// the exporter can wait before uploading view data to
+	// the backend.
+	// Optional.
+	ExportDelayThreshold time.Duration
+
+	// ExportCountThreshold determines how many view data events
+	// can be buffered before batch uploading them to the backend.
+	// Optional.
+	ExportCountThreshold int
+
+	bundler *bundler.Bundler
+
+	measuresMu sync.Mutex
+	measures   map[stats.Measure]struct{} // measures already created remotely
+
+	once sync.Once
+	c    *monitoring.MetricClient
+}
+
+// Export exports to the Stackdriver Monitoroing if view data
+// has one or more rows.
+func (e *Exporter) Export(vd *stats.ViewData) {
+	e.once.Do(e.newClient)
+	if len(vd.Rows) == 0 {
+		return
+	}
+	e.bundler.Add(vd, 1)
+}
+
+func (e *Exporter) onError(err error) {
+	if e.OnError != nil {
+		e.OnError(err)
+		return
+	}
+	log.Printf("Failed to export to Stackdriver Monitoring: %v", err)
+}
+
+func (e *Exporter) newClient() {
+	client, err := monitoring.NewMetricClient(context.Background(), e.ClientOptions...)
+	if err != nil {
+		e.OnError(err)
+		return
+	}
+	e.c = client
+	e.measures = make(map[stats.Measure]struct{})
+	e.bundler = bundler.NewBundler((*stats.ViewData)(nil), func(bundle interface{}) {
+		vds := bundle.([]*stats.ViewData)
+		if err := e.upload(vds); err != nil {
+			e.onError(err)
+		}
+	})
+	e.bundler.DelayThreshold = e.ExportDelayThreshold
+	e.bundler.BundleCountThreshold = e.ExportCountThreshold
+}
+
+func (e *Exporter) upload(vds []*stats.ViewData) error {
+	ctx := context.Background()
+
+	// TODO(jbd): Create measures concurrently?
+	for _, vd := range vds {
+		if err := e.createMeasure(ctx, vd); err != nil {
+			return err
+		}
+	}
+	if err := e.c.CreateTimeSeries(ctx, e.makeReq(vds)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (e *Exporter) makeReq(vds []*stats.ViewData) *monitoringpb.CreateTimeSeriesRequest {
+	timeSeries := make([]*monitoringpb.TimeSeries, len(vds))
+	for i, vd := range vds {
```

Comment:
I think each `Row` in a `ViewData` should be converted to one `TimeSeries` (i.e one `ViewData` -> a list of `TimeSeries`), because one `TimeSeries` can only have one set of labels (tags).

exporter in Java: https://github.com/census-instrumentation/opencensus-java/pull/713/files#diff-d2886f5b11a5e3be7a00a8f5fff5a1f0R148